src/CoreFoundation/CFArray.cs

```diff
@@ -85,19 +85,12 @@ protected virtual void Dispose (bool disposing)
 		}
 
 		// pointer to a const struct (REALLY APPLE?)
-		static readonly IntPtr kCFTypeArrayCallbacks_ptr;
-
-		// FIXME: right now we can't use [Fields] for GetIndirect
-		static CFArray ()
-		{
-			var handle = Dlfcn.dlopen (Constants.CoreFoundationLibrary, 0);
-			if (handle == IntPtr.Zero)
-				return;
-			try {
-				kCFTypeArrayCallbacks_ptr = Dlfcn.GetIndirect (handle, "kCFTypeArrayCallBacks");
-			}
-			finally {
-				Dlfcn.dlclose (handle);
+		static IntPtr kCFTypeArrayCallbacks_ptr_value;
+		static IntPtr kCFTypeArrayCallbacks_ptr {
+			get {
+				if (kCFTypeArrayCallbacks_ptr_value == IntPtr.Zero)
+					kCFTypeArrayCallbacks_ptr_value = Dlfcn.GetIndirect (Libraries.CoreFoundation.Handle, "kCFTypeArrayCallBacks");
+				return kCFTypeArrayCallbacks_ptr_value;
```

Comment:
Why not move this into `corefoundation.cs` as `[Field]` ?
That will give us some *free** introspection and xtro tests and less manual code to maintain.